packages/components/src/utils/focusTrap.ts

```diff
@@ -42,26 +47,42 @@ export const setFocusTrap = (
   let focusableElements: FirstAndLastFocusableElement = [];
 
   document.removeEventListener('keydown', documentKeydownListener);
+
   if (isOpen) {
-    focusableElements = getFirstAndLastFocusableElement(host, closeBtn);
+    // the filter removes falsy values: https://stackoverflow.com/a/41346932
+    focusableElements = getFirstAndLastFocusableElement(host, closeBtn).filter((x) => x);
```

Comment:
Not sure if we ever had a problem with this but if yes, then unit test covering/reproducing this is missing.
I personally would revert all these changes, though, since all the early returns make it more difficult to read and you have to keep all the opposites, e.g. `key === 'Tab'` in mind to understand what is going on. This was clearer before and could have been extended with comments if necessary.

Reply:
I reverted it

Reply:
The refactoring helped me understand it since it was nested if/else before which is, at least for my opinion, harder to understand but the refactoring adds no business value and we want to get rid of the focusTrap at all in the future so I will revert it.